src/gui/properties/downloadedpiecesbar.h

```diff
@@ -32,68 +32,39 @@
 #define DOWNLOADEDPIECESBAR_H
 
 #include <QWidget>
-#include <QPainter>
-#include <QImage>
 #include <QBitArray>
 #include <QVector>
 
-class QHelpEvent;
+#include "piecesbar.h"
 
-namespace BitTorrent
+class DownloadedPiecesBar: public PiecesBar
 {
-    class TorrentHandle;
-}
-
-class DownloadedPiecesBar: public QWidget {
     Q_OBJECT
-    Q_DISABLE_COPY(DownloadedPiecesBar)
```

Comment:
Repeat the macro because of https://stackoverflow.com/questions/19854371/repeating-q-disable-copy-in-qobject-derived-classes